modules/Bio/EnsEMBL/Compara/RunnableDB/CheckGenomeReusability.pm

```diff
@@ -197,23 +189,7 @@ sub run {
 
     return if(defined($self->param('reuse_this')));  # bypass run() in case 'reuse_this' has either been passed or already computed
 
-    my ($prev_hash, $curr_hash);
-    if (comes_from_core_database($self->param('genome_db'))) {
-        $prev_hash = hash_all_exons_from_dbc( $self->param('prev_core_dba') );
-        $curr_hash = hash_all_exons_from_dbc( $self->param('curr_core_dba') );
-    } else {
-        $prev_hash = hash_all_sequences_from_db( $self->param('reuse_genome_db') );
-        $curr_hash = hash_all_sequences_from_file( $self->param('genome_db') );
-    }
-    my ($removed, $remained1) = check_hash_equals($prev_hash, $curr_hash);
-    my ($added, $remained2)   = check_hash_equals($curr_hash, $prev_hash);
-
-    my $coding_objects_differ = $added || $removed;
-    if($coding_objects_differ) {
-        $self->warning("The coding objects changed: $added hash keys were added and $removed were removed");
-    }
-
-    $self->param('reuse_this', $coding_objects_differ ? 0 : 1);
+    $self->param('reuse_this', $self->run_comparison());
```

Comment:
Maybe this method could be added in a comment somehow so everybody knows it needs to be implemented by each subclass?